packages/-ember-data/tests/integration/relationships/has-many-test.js

```diff
@@ -4095,4 +4088,117 @@ module('integration/relationships/has_many - Has-Many Relationships', function (
       });
     });
   });
+
+  test('fetch records with chained async has-many, ensure the leafs are retrieved', async function (assert) {
+    assert.expect(4);
+
+    let store = this.owner.lookup('service:store');
+    let adapter = store.adapterFor('application');
+
+    store.modelFor('user').reopen({
+      messages: hasMany('message', { polymorphic: true, async: true }),
```

Comment:
why do we use `polymorphic: true` here? at least in crates.io we don't seem to have any polymorphic relationships 🤔

Reply:
just because this is how the relation is defined at the top of the file.  (https://github.com/emberjs/data/blob/df969d59ca2eb0912dcccfb7d59b33332e3a9d55/packages/-ember-data/tests/integration/relationships/has-many-test.js#L28)

I wanted to use the 'same' relation, but async.